client/src/helpers/requestApi.js

```diff
@@ -0,0 +1,29 @@
+export default async function requestApi(url, method, data = null) {
+  try {
+    const headers = new Headers();
+    headers.append('Authorization', `Bearer ${localStorage.getItem('token')}`);
+    headers.append('Origin', 'http://localhost:3000');
+
+    let body;
+    if (data) {
+      headers.append('Content-Type', 'application/json');
+      body = JSON.stringify(data);
+    }
+
+    const response = await fetch(`${process.env.REACT_APP_API_URL}${url}`, {
+      method,
+      headers,
+      body,
+    });
+
+    if (!response.ok) {
+      console.log(response.error);
+    }
+
+    const result = await response.json();
+    return result;
+  } catch (error) {
+    console.log(error);
+    return error;
```

Comment:
i'm not sure that you should return the error in the way like result;
you should be able to differ them.

it's better to return a  { data, error } object, where error is null if the request was succeeded, data is null otherwise